### ISSUE TITLE
Samsung Galaxy j7 duo 2018 j720f - silent death with Magisk over night if charging

### ISSUE THREAD

**Issue by szmaszmo1** [topjohnwu/Magisk#1597]
After installing Magisk the phone dies over night if put to charging e.g. by 50% battery level ( reset with buttons or pulling battery is needed). Maybe it has nothing to do with charging, but.. during the day everything is fine (even if charging). I suspect it has to do with phone idle state.
Since TWRP is not available (or I can not find it) first I tried using https://forum.xda-developers.com/galaxy-j7/help/sumsung-j7-duo-sm-j720f-recovery-root-t3843618 using J720FDDU1ARCJ (first available oreo firmware from Samsung).
Uninstalling Magisk solved the problem. 
Flashed again to original J720FDDS4ASE1 - also no problem.
![sammobile_j720f_available_ins_firmware_by_date](https://user-images.githubusercontent.com/51872904/59560531-d8f19480-9013-11e9-822e-983f18fe5f6c.png)

Now I tried the newest one J720FDDS4ASE1 following official for new 2019 devices (flashed via odin CP(org)+BL(org)+AP(magisk_patched.tar)+CSC(org)) https://github.com/topjohnwu/Magisk/blob/master/docs/install.md steps.
The same problem occurs again.
All logs I managed to collect after night are here:
[data.tar.gz](https://github.com/topjohnwu/Magisk/files/3293893/data.tar.gz)
Kernel logs after reset disappear.. so I have no idea how to better trace it.
Magisk is currently on Canary (debug) update channel.
If you need other logs, or other testcases please let me know.
Thanks for your effort.

**Comment by topjohnwu** [comment]
This doesn't seem to have anything to do with Magisk

**Comment by szmaszmo1** [comment]
Thank you for your response, however I am still not convinced as you did not provide any arguments.
In meantime, I have traced the device by logcat -f ./log.txt and this is the relevant output:
<pre>
06-19 03:00:24.843  3312 20259 D vold    : Starting trim of /cache
06-19 03:00:24.844  5363  5363 I ApplicationPackageManager: load= live icon for com.sec.android.app.clockpackage, from overlay = false
06-19 03:00:24.845  5363  5363 I LiveIconManager: getLiveIcon: complete(sync_created)
06-19 03:00:24.846  5363  5363 I IconView: applyFromShortcutInfo - end GetLive : com.sec.android.app.clockpackage
06-19 03:00:24.847  5363  5363 I IconView: applyFromApplicationInfo - start GetLive : com.sec.android.app.clockpackage
06-19 03:00:24.847  5363  5363 I IconView: applyFromApplicationInfo - end GetLive : com.sec.android.app.clockpackage
06-19 03:00:24.847  5363  5363 I IconView: applyFromApplicationInfo - start GetLive : com.samsung.android.calendar
06-19 03:00:24.847  5363  5363 I IconView: applyFromApplicationInfo - end GetLive : com.samsung.android.calendar
06-19 03:00:24.870  3312 20259 I vold    : Trimmed 291536896 bytes on /cache in 25ms
06-19 03:00:24.870  3312 20259 D vold    : Starting trim of /data
06-19 03:00:24.871  3996  4130 D VoldConnector: RCV <- {662 /cache 291536896 25163500}
06-19 03:00:24.871  4366  4366 D KeyguardUpdateMonitor: handleStatusBarState( false )
06-19 03:00:24.872  3996  4021 I StorageManagerService: onEvent:: raw= 662 /cache 291536896 25163500 cooked =  662 /cache 291536896 25163500
06-19 03:00:24.886  3996  4014 W BroadcastQueue: Background execution not allowed: receiving Intent { act=android.intent.action.DROPBOX_ENTRY_ADDED flg=0x10 (has extras) } to com.google.android.gms/.stats.service.DropBoxEntryAddedReceiver
06-19 03:00:24.887  3996  4014 W BroadcastQueue: Background execution not allowed: receiving Intent { act=android.intent.action.DROPBOX_ENTRY_ADDED flg=0x10 (has extras) } to com.google.android.gms/.chimera.GmsIntentOperationService$PersistentTrustedReceiver
06-19 03:00:25.011  3996  4006 D ConnectivityService: filterNetworkStateForUid() uid: 10027 networkInfo: [type: WIFI[] - WIFI, state: CONNECTED/CONNECTED, reason: (unspecified), extra: "UPC4B622F9", failover: false, available: true, roaming: false, metered: false]
</pre>
Further on, I discovered that the command which cause the phone to partial freeze (I can still ping device or use sshd session) is:
<pre>
vdc fstrim dotrim
</pre>
It is worth to mention, that trimmer app (https://play.google.com/store/apps/details?id=com.fifthelement.trimmer&hl=en) does NOT cause the issue (also doing fstrim of /data and /cache).
The connection between vold/vdc commands and Magisk is visible by mount command:
<pre>
/sbin/.magisk/block/system on /system/bin/vdc type ext4 (ro,seclabel,relatime)
/sbin/.magisk/block/system on /system/bin/vold type ext4 (ro,seclabel,relatime)
</pre>
Since there is no TWRP and/or recovery mode for this device available (or I can not find it), I used yet another app eMMC Brick Bug Check (https://play.google.com/store/apps/details?id=net.vinagre.android.emmc_check&hl=en) instead of efsck in order to check device memory.
There is no evidence of any NAND erros. By flashing, Odin also did not report any NAND errors. 

So in my opinion, either I do some mistake by Magisk flashing procedure or it is a bug in Magisk, or it is a bug in Android OS triggered by Magisk installation.
Since I can not easily control vold daemon, the only workaround, which I can imagine from my side would be to add "notrim" flag to sth. like fstab in android (which does not seem to be straight forward task).  Any other ideas, hints?
